### PR TITLE
listen for missing stake errors and add stake flag to user profiles

### PR DESCRIPTION
**Pull request by 0zzy-o**

### CODE REVIEW

File: src/graphql/schema.js [hunk 1]
```diff
@@ -81,6 +82,7 @@ input ProfileInput {
   avatar: String
   name: String
   bio: String
+  hasStake: Boolean
```
Comment (PaulLeCam) [inline src/graphql/schema.js]: Does this need to be added to `ProfileInput`? The server shouldn't rely on the client to provide this.

File: src/pss/client.js [hunk 2]
```diff
@@ -147,12 +147,21 @@ export const joinChannelTopic = async (
...
+        await pss.setPeerPublicKey(p.pubKey, channel.topic, p.address)
+        logClient('add peer', channel.topic, p.pubKey)
+        topic.addPeer(p.pubKey)
+        if (!contact.profile.hasStake) {
+          db.setContactStake(p.pubKey, true)
+        }
+      } catch (err) {
+        if (err.message.includes('No stake found')) {
+          db.setContactStake(p.pubKey, false)
+        }
```
Comment (PaulLeCam) [inline src/pss/client.js]: It probably needs to throw the error if it's another one than the staking, ex
```js
if (...) { // existing logic
  ...
} else {
  throw err
}

File: src/pss/client.js [hunk 3]
```diff
@@ -461,7 +482,10 @@ export const addContactRequest = async (
   payload: ContactRequestPayload,
 ) => {
   const contact = {
-    profile: payload.profile,
+    profile: {
+      ...payload.profile,
+      hasStake: true,
```
Comment (PaulLeCam) [inline src/pss/client.js]: Any reason `hasStake` is set to `true` here without actually checking the stake?

Comment (0zzy-o) [reply src/pss/client.js]: As this is from when you receive a contact request I thought we could presume the user has a stake to be able to make the request in the first place?

File: src/graphql/schema.js (hunk 1 above)
Comment (0zzy-o) [reply src/graphql/schema.js]: was a way to flag hasStake on users own profile but you're right probably better to not have client handle this

File: src/pss/client.js (hunk 3 above)
Comment (PaulLeCam) [reply src/pss/client.js]: I don't think we should trust peers to provide this information, if they are using the same version of the server that works, but for example a contact request sent from the current version of onyx-server would make this code accept contacts even when they don't have a stake.

**Review by PaulLeCam — APPROVED** [review]

**Pull request merged**